core/src/main/scala/org/apache/spark/storage/StorageStatusListener.scala

```diff
@@ -88,4 +88,19 @@ class StorageStatusListener extends SparkListener {
     }
   }
 
+  override def onBlockUpdate(blockUpdateEvent: SparkListenerBlockUpdate) = synchronized {
+    val executorId = blockUpdateEvent.blockManagerId.executorId
+    updateStorageStatus(executorId, Seq((blockUpdateEvent.blockId,
```

Comment:
If I'm following things correctly, this will also get called for RDD blocks, which are already getting updated by `TaskEnd`.  I suppose its not a problem for them to get updated again with the same info, but you could probably skip the update for non-broadcast blocks, and include a comment as in `EventLoggingListener`.